lrauv_ignition_plugins/src/HydrodynamicsPlugin.cc

```diff
@@ -247,8 +247,7 @@ void HydrodynamicsPlugin::PreUpdate(
   // Since we are transforming angular and linear velocity we only care about
   // rotation
   auto localLinearVelocity = pose->Rot().Inverse() * linearVelocity->Data();
-  auto localRotationalVelocity = pose->Rot().Inverse() * *rotationalVelocity /
-    (2 * IGN_PI);
+  auto localRotationalVelocity = pose->Rot().Inverse() * *rotationalVelocity;
```

Comment:
This looks like a bugfix similar to the added mass on #59 . Another reason we need unit tests for the hydrodynamics :confused:

Reply:
This would be great to isolate it into a separate PR since this is unrelated to this change.

Reply:
This is very much related to the current pr. Without the change in units, the behaviour of the model would result in oscillations. The problem at hand here was that the units of the coefficients were wrong. Since these coefficients are nonlinear, there is no way to scale them.

Reply:
It's related in that it's used in the same unit tests that need to pass. But this PR is stated as adjusting the water density. Developing this PR exposed the bug, but the bug exists on main and should be able to be resolved upstream independent of this. It's here and we should probably just roll with it. But in cases like this it can significantly help us with our review process/velocity in the future to notice this. Pull it out and submit it in a parallel PR. Things like this may effect other PRs beyond the one that it was found when debugging. If we keep these smaller bug fixes separate we can merge them quickly and not end up with very long running high overhead PRs which take a long time to QA and merge.

Reply:
OK my bad I thought I had added this to the PR description but seem to have deleted the bullet. Unfortunately I cannot agree with splitting this out because if I were to do so it would mean having to re-tweak the unit tests. Our dynamics on main are broken, and the unit tests there do not actually check what we need to be checked (matching trajectories is very fragile). Any change in dynamics therefore means that I have to go back and recheck those unit tests. This results in a loop where we spend ages trying to get the unit tests to pass with fudged tolerances. These high tolerances are equivalent to just commenting out the unit tests as it means we do not check the behavior at all. So while I agree with small PRs, I would not actually want to split this out onto a different PR just to have to tweak tolerances over and over again.

Reply:
I ticketed https://github.com/ignitionrobotics/ign-gazebo/issues/1273 upstream for us to add tests for the hydrodynamics plugin. Ideally this kind of change would come with a test that clearly defines the before and after behaviour.